public/plugins/datasources/datasource_plugin/datasource_plugin.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DataExplorerPlugin, IPluginMeta } from '../../data_explorer/data_explorer_plugin';
+
+export type IDatasourcePluginMeta = IPluginMeta;
+
+export interface IDatasourcePluginComponents {
+  QueryEditor: any;
```

Comment:
React.ReactNode?

Reply:
Yes, later I'll compose more specific component type with props, this is in progress, but will add such type